my_Enumerable.rb

```diff
@@ -0,0 +1,153 @@
+  module Enumerable
+    def my_each
+      # your code here
+      return to_enum(:double) unless block_given?
+      for i in 0..self.length-1
+        yield(self[i])
+      end
+      return self
+    end
+
+    def my_each_with_index
+      return to_enum(:double) unless block_given?
+      for i in 0..self.length-1
+        yield(self[i],i)
+      end
+      return self
+    end
+
+    def my_select
+      return to_enum(:double) unless block_given?
+      conditionMet = []
+      self.my_each do |item|
+        if yield(item)
+          conditionMet << item
+        end
+      end
+      conditionMet
+    end
+
+    def my_all?(pattern = nil)
+      self.my_each do |item|
+        if block_given?
+          unless yield(item)
+            return false
+          end
+        else
+          if pattern
+            unless pattern === (item)
+              return false
+            end
+          else
+            if item == nil||item == false
+              return false
+            end
+          end
+        end
+      end
+      return true
+    end
+
+    def my_any?(pattern = nil)
+      self.my_each do |item|
+        if block_given?
+          if yield(item)
+            return true
+          end
+        else
+          if pattern
+            if pattern === (item)
+              return true
+            end
+          else
+            if item == nil||item == false
+              return true
+            end
+          end
+        end
+      end
+      return false
+    end
+
+    def my_none?(pattern = nil)
+      self.my_each do |item|
+        if block_given?
+          if yield(item)
+            return false
+          end
+        else
+          if pattern
+            if pattern === (item)
+              return false
+            end
+          else
+            if item == nil||item == false
+              return false
```

Comment:
### 2. Similar problem as the one seen in my_any? , expected behavior compared with the current method implemented in your project:

```
irb(main):174:0> arr = [false, false]
=> [false, false]
irb(main):175:0> arr.none?
=> true
irb(main):176:0> arr.my_none?
=> false

```